skyve-war/src/main/java/modules/admin/SelfRegistration/actions/Register.java

```diff
@@ -63,35 +60,48 @@ public ServerSideActionResult<SelfRegistrationExtension> execute(SelfRegistratio
-					List<String> countryList = Arrays.asList(UtilImpl.COUNTRY_CODES.split("//|"));
-					// Is country on list
-					boolean found = countryList.stream()
-							.anyMatch(s -> s.equalsIgnoreCase(countryCode.get()));
-					if (found) {
-						// Check if the list is a blacklist and ban the country if it is
-						if(UtilImpl.COUNTRY_LIST_TYPE.equalsIgnoreCase(BLACK_LIST)) {
-							LOGGER.warn(
-									"Self-registration failed because country was on the blacklist. Suspect bot submission for "
-											+ bean.getUser().getContact().getName() + ", " + bean.getUser().getContact().getEmail1());
-							bean.setPassSilently(Boolean.TRUE);
-							return new ServerSideActionResult<>(bean);
-						}
-					} else if(!found) {
-						// Check if the list is a white list and ban the country if the list is a white list
-						if(UtilImpl.COUNTRY_LIST_TYPE.equalsIgnoreCase(WHITE_LIST)) {
-							LOGGER.warn(
-									"Self-registration failed because country was not on the whitelist. Suspect bot submission for "
-											+ bean.getUser().getContact().getName() + ", " + bean.getUser().getContact().getEmail1());
-							bean.setPassSilently(Boolean.TRUE);
-							return new ServerSideActionResult<>(bean);
+			// If configured, check the country and if it is on the blacklist/whitelist
+			if (UtilImpl.IP_INFO_TOKEN != null) {
+				HttpServletRequest request = EXT.getHttpServletRequest();
+				String clientIpAddress = SecurityUtil.getSourceIpAddress(request);
+				LOGGER.info("Checking country for ip " + clientIpAddress);
+				Optional<String> countryCode = geoIpService.getCountryCodeForIp(clientIpAddress);
+				if (countryCode.isPresent()) {
+					String country = countryCode.get();
+					LOGGER.info("Registration request from country " + country);
+					if (UtilImpl.COUNTRY_CODES != null) {
+						List<String> countryList = Arrays.asList(UtilImpl.COUNTRY_CODES.split("\\|"));
+						// Is this country on the list?
+						boolean found = countryList.stream()
+								.anyMatch(s -> s.equalsIgnoreCase(country));
+						if (found) {
+							// Check if the list type is a blacklist and ignore the registration if this country is on the deny list
+							if (UtilImpl.COUNTRY_LIST_TYPE.equalsIgnoreCase(CountryListType.blacklist.name())) {
+								String message = "Self-registration failed because country " + country
+										+ " is on the blacklist. Suspected bot submission for "
+										+ bean.getUser().getContact().getName() + " - " + bean.getUser().getContact().getEmail1();
+								LOGGER.warn(message);
+								
+								// Record security event
+								SecurityUtil.log("GEO IP Block", message);
+								
+								bean.setPassSilently(Boolean.TRUE);
+								return new ServerSideActionResult<>(bean);
+							}
+						} else {
+							// Check if the list type is a white list and ignore the registration if this country is not on the allow list
+							if (UtilImpl.COUNTRY_LIST_TYPE.equalsIgnoreCase(CountryListType.whitelist.name())) {
```

Comment:
Should this be a not equals check?

Reply:
No - the flow defined here is if a users country is not on the list **AND** the list is a whitelist, fail.